examples/advanced/quantum_volume.ipynb

```diff
@@ -0,0 +1,254 @@
+      "text/plain": [
+       "<Figure size 432x288 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "from examples.advanced import quantum_volume\n",
+    "import numpy as np\n",
+    "from matplotlib import pyplot as plt\n",
+    "from matplotlib.ticker import FormatStrFormatter\n",
+    "\n",
+    "# Some set-up variables. Feel free to play with these.\n",
+    "num_repetitions = 10 # This is supposed to be >= 100.\n",
+    "depths = range(2, 5) # The depths and number of qubits\n",
+    "\n",
+    "# Keep track of the ideal and noisy probabilities.\n",
+    "average_ideals = []\n",
+    "average_noisies = []\n",
+    "\n",
+    "for depth in depths:\n",
+    "    num_qubits = depth # Square matrix.\n",
+    "    print(\"Running simulation with %s qubits and a depth of %s\" % (num_qubits, depth))\n",
+    "    average_ideal = 0\n",
+    "    average_noisy = 0\n",
+    "    for i in range(num_repetitions):\n",
+    "        print(\"    Repetition %s of %s\" % (str(i + 1), str(num_repetitions)))\n",
+    "        # Generate a model circuit.\n",
+    "        model_circuit = quantum_volume.generate_model_circuit(\n",
+    "            num_qubits, depth, random_state=np.random.RandomState()\n",
+    "        )\n",
+    "        # Compute its heavy set.\n",
+    "        heavy_set = quantum_volume.compute_heavy_set(model_circuit)\n",
+    "        print(\"        Heavy Set:\", heavy_set)\n",
+    "        \n",
+    "        # Run an ideal simulator over the model circuit.\n",
+    "        ideal = quantum_volume.sample_heavy_set(model_circuit, heavy_set)\n",
+    "        average_ideal += ideal\n",
+    "        print(\"        Ideal simulation HOG probability: \", ideal)\n",
+    "\n",
+    "        # Run a noisy simulator over the model circuit.\n",
+    "        noisy = quantum_volume.sample_heavy_set(model_circuit, heavy_set, sampler=cirq.DensityMatrixSimulator(noise=cirq.ConstantQubitNoiseModel(qubit_noise_gate=cirq.DepolarizingChannel(p=0.005))))\n",
+    "        print(\"        Noisy simulation HOG probability: \", noisy)\n",
+    "        average_noisy += noisy\n",
+    "        \n",
+    "        # TODO: The next steps are to compile the model circuit and run it on various devices.\n",
+    "\n",
+    "    # Compute the average performance over the total number of runs.\n",
+    "    average_ideal /= num_repetitions\n",
+    "    average_noisy /= num_repetitions\n",
+    "    print(\"    Average ideal HOG probability:\", average_ideal)\n",
+    "    print(\"    Average noisy HOG probability:\", average_noisy)\n",
+    "    average_ideals.append(average_ideal)\n",
+    "    average_noisies.append(average_noisy)\n",
+    "\n",
+    "# Create the chart. Note that we try to make it look as similar\n",
```

Comment:
Might be clearer if extracted into a method.

Is the call to `plt.show()` required?

Reply:
Done, and no it was not, so I removed.